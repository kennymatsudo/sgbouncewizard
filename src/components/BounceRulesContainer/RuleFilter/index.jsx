import React from "react";
import { TextInput } from "@sendgrid/ui-components/text-input";
import { Select } from "@sendgrid/ui-components/select";
import { Row } from "../../Row";
import { Column } from "../../Column";
import "./index.scss";

const RuleFilter = ({
  searchToken,
  updateSearchToken,
  updateSearchCategory,
  filterOptions,
  addFilter,
  removeFilter,
  invalidFilter,
}) => (
  <div className="filter-wrap">
    <div className="filter-header">
      <p className="filter-title">
        {"Search bounce rules by "}
        <a className="filter-option-toggle" href="#">
          all
        </a>
        {" of the following:"}
      </p>
      <div className="btn-list">
        <button type="button" className="btn btn-small btn-secondary">
          Clear Search
        </button>
      </div>
    </div>
    <div className="filter-list">
      <Row>
        <Column width={3} offset={1}>
          <div style={{ paddingTop: "15px" }} />
          <Select
            onChange={updateSearchCategory}
            name="primary-filter"
            defaultValue={{ label: "Bounce Action", value: "Bounce Action" }}
            options={[
              { label: "Bounce Action", value: "Bounce Action" },
              { label: "Enhanced Code", value: "Enhanced Code" },
              { label: "Description", value: "description" },
            ]}
          />
        </Column>
        <Column width={9} offset={4}>
          <TextInput
            type="text"
            fullWidth
            label="Description"
            onChange={updateSearchToken}
            value={searchToken}
            isValid={!invalidFilter}
            info={invalidFilter && "Please enter a valid filter description."}
          />
        </Column>
        <i className="filter-row-remove sg-icon sg-icon-x" />
      </Row>

      <Row>
        <div className="row btn-list filter-actions-container">
          <button
            type="button"
            onClick={addFilter}
            className="btn btn-small btn-secondary"
          >
            Add a Filter
          </button>
          {filterOptions.map(filterOption => (
            <button
              type="button"
              className="btn btn-small btn-secodary"
              onClick={removeFilter}
              key={JSON.stringify(filterOption)}
            >
              {filterOption.searchCategory} - {filterOption.searchToken}
            </button>
          ))}
        </div>
      </Row>
    </div>
  </div>
);
export default RuleFilter;
