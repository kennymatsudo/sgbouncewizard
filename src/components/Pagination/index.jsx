import React from "react";
import cn from "classnames";
import "./index.scss";

const pagesToDisplay = 5;

const Pagination = ({
  prevPageIndex,
  nextPageIndex,
  updatePageIndex,
  pageIndex,
  pageInterval,
  numRules,
}) => {
  const endPage =
    pageIndex +
    (pagesToDisplay -
      (pageIndex % pagesToDisplay === 0
        ? pagesToDisplay
        : pageIndex % pagesToDisplay));
  const startPage = endPage - (pagesToDisplay - 1);
  const totalPages = Math.ceil(numRules / pageInterval);
  return (
    <div className="pagination pagination-container">
      <a
        className={`btn btn-secondary btn-small pagination-prev ${
          pageIndex <= pagesToDisplay ? "is-disabled" : ""
        }`}
        onClick={prevPageIndex}
        onKeyDown={() => prevPageIndex}
        role="button"
        tabIndex="0"
      >
        Prev
      </a>
      {pageIndex > pagesToDisplay && (
        <span>
          <a
            className="pagination-link"
            onClick={() => updatePageIndex(1)}
            onKeyDown={() => updatePageIndex(1)}
            role="button"
            tabIndex="0"
          >
            {1}
          </a>
          <a className="pagination-ellipses">&hellip;</a>
        </span>
      )}
      <div className="pagination-links">
        {Array(endPage - startPage + 1)
          .fill()
          .map((_, i) => startPage + i)
          .map(number => (
            <a
              key={number}
              className={cn("pagination-link", {
                "is-active": number === pageIndex,
              })}
              onClick={() => updatePageIndex(number)}
              onKeyDown={() => updatePageIndex(number)}
              role="button"
              tabIndex="0"
            >
              {number}
            </a>
          ))}
        {pageIndex <= totalPages - pagesToDisplay && (
          <span>
            <a className="pagination-ellipses">&hellip;</a>
            <a
              className="pagination-link"
              onClick={() => updatePageIndex(totalPages)}
              onKeyDown={() => updatePageIndex(totalPages)}
              role="button"
              tabIndex="0"
            >
              {totalPages}
            </a>
          </span>
        )}
      </div>
      <a
        className={`btn btn-secondary btn-small pagination-next ${
          pageIndex > totalPages - pagesToDisplay ? "is-disabled" : ""
        }`}
        onClick={nextPageIndex}
        onKeyDown={() => nextPageIndex}
        role="button"
        tabIndex="0"
      >
        Next
      </a>
    </div>
  );
};

export default Pagination;
