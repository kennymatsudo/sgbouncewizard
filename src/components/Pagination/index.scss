.pagination-container {
  margin-bottom: 30px;
}
