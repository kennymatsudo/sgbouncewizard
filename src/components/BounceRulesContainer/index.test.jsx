import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import BounceRuleContainer, { RuleListContainer } from ".";
import RuleFilter from "../RuleFilter";
import Pagination from "../Pagination";

const wrapper = shallow(
  <BounceRuleContainer
    filteredRules={[
      {
        id: 504,
        response_code: "550",
        enhanced_code: "",
        regex: "no MX record for domain",
        priority: 0,
        description:
          "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
        bounce_action: "no_action",
      },
    ]}
    filterOptions={[]}
    pageIndex={1}
  />
);

it("should render correctly", () => {
  const tree = renderer
    .create(
      <BounceRuleContainer
        rules={[
          {
            id: 504,
            response_code: "550",
            enhanced_code: "",
            regex: "no MX record for domain",
            priority: 0,
            description:
              "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
            bounce_action: "no_action",
          },
        ]}
        filteredRules={[
          {
            id: 504,
            response_code: "550",
            enhanced_code: "",
            regex: "no MX record for domain",
            priority: 0,
            description:
              "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
            bounce_action: "no_action",
          },
        ]}
        filterOptions={[]}
        pageIndex={2}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render a create a bounce rule", () => {
  expect(wrapper.find('[data-test="create-rule-button"]')).toHaveLength(1);
});

it("should render a export to csv button", () => {
  expect(wrapper.find('[data-test="export-csv-button"]')).toHaveLength(1);
});

it("should render a filter component", () => {
  expect(wrapper.find(RuleFilter)).toHaveLength(1);
});

it("should render a rule list component", () => {
  expect(wrapper.find(RuleListContainer)).toHaveLength(1);
});

it("should render paginiation", () => {
  expect(wrapper.find(Pagination)).toHaveLength(1);
});
