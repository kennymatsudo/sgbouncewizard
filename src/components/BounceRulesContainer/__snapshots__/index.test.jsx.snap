// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container"
  data-test="bounce-rules-page"
>
  <Header
    name="Kenny"
  />
  <Component>
    <Component
      offset={2}
      width={6}
    >
      <Component
        withoutTrailingSlash={false}
      >
        <a
          href="/bounce_rules"
        >
          Bounce Rules
        </a>
      </Component>
    </Component>
  </Component>
  <Component>
    <Component
      offset={2}
      width={2}
    >
      <h1>
        Bounce Rules
      </h1>
    </Component>
    <Component
      className="csv-button-col"
      offset={10}
      width={1}
    >
      <Component
        active={false}
        badge={null}
        className="csv-button"
        data-test="export-csv-button"
        disabled={false}
        group={false}
        icon={null}
        loading={false}
        onClick={[Function]}
        onDark={false}
        small={false}
        type="secondary"
      >
        Export CSV
      </Component>
    </Component>
    <Component
      offset={11}
      width={1}
    >
      <div
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        <Component
          active={false}
          badge={null}
          className="create-rule-button"
          data-test="create-rule-button"
          disabled={false}
          group={false}
          icon={null}
          loading={false}
          onClick={[Function]}
          onDark={false}
          small={false}
          type="primary"
        >
          Create Rule
        </Component>
      </div>
    </Component>
  </Component>
  <Component>
    <Component
      offset={2}
      width={10}
    >
      <RuleFilter
        filterOptions={Array []}
      />
    </Component>
  </Component>
  <Component>
    <Component
      offset={2}
      width={10}
    >
      <RuleListContainer
        rules={
          Array [
            Object {
              "bounce_action": "no_action",
              "description": "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
              "enhanced_code": "",
              "id": 504,
              "priority": 0,
              "regex": "no MX record for domain",
              "response_code": "550",
            },
          ]
        }
      />
    </Component>
  </Component>
  <Component>
    <Component
      offset={5}
      width={4}
    >
      <Pagination
        pageIndex={1}
      />
    </Component>
  </Component>
  <SideModal
    bodyNode={<body />}
    modalContainer={<body />}
  >
    <div
      className="create-rule-modal"
    >
      <div
        data-test="rule-filter"
      >
        <div
          className="filter-wrap"
        >
          <div
            className="filter-header"
          >
            <p
              className="filter-title"
            >
              Search emails by
               
              <a
                className="filter-option-toggle"
                href="#"
              >
                all
              </a>
               
              of the following:
            </p>
            <div
              className="btn-list"
            >
              <button
                className="btn btn-small btn-secondary"
                type="button"
              >
                Clear Search
              </button>
            </div>
          </div>
          <div
            className="filter-list"
          >
            <div
              className="mako-grid__row"
            >
              <div
                className="mako-grid__col-3 col-start-1"
              >
                <div
                  style={
                    Object {
                      "paddingTop": "15px",
                    }
                  }
                />
                <div
                  className="input-select-wrap"
                >
                  <div
                    className="css-10nd86i"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="css-12nb859"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="css-1m8oe7c"
                      >
                        <div
                          className="css-1v8euj5"
                        >
                          Bounce Action
                        </div>
                        <div
                          className="css-1g6gooi"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-2-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              theme={
                                Object {
                                  "borderRadius": 4,
                                  "colors": Object {
                                    "danger": "#DE350B",
                                    "dangerLight": "#FFBDAD",
                                    "neutral0": "hsl(0, 0%, 100%)",
                                    "neutral10": "hsl(0, 0%, 90%)",
                                    "neutral20": "hsl(0, 0%, 80%)",
                                    "neutral30": "hsl(0, 0%, 70%)",
                                    "neutral40": "hsl(0, 0%, 60%)",
                                    "neutral5": "hsl(0, 0%, 95%)",
                                    "neutral50": "hsl(0, 0%, 50%)",
                                    "neutral60": "hsl(0, 0%, 40%)",
                                    "neutral70": "hsl(0, 0%, 30%)",
                                    "neutral80": "hsl(0, 0%, 20%)",
                                    "neutral90": "hsl(0, 0%, 10%)",
                                    "primary": "#2684FF",
                                    "primary25": "#DEEBFF",
                                    "primary50": "#B2D4FF",
                                    "primary75": "#4C9AFF",
                                  },
                                  "spacing": Object {
                                    "baseUnit": 4,
                                    "controlHeight": 38,
                                    "menuGutter": 8,
                                  },
                                }
                              }
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="css-1wy0on6"
                      >
                        <span
                          className="css-0"
                        />
                        <div
                          aria-hidden="true"
                          className="css-86sq3t"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-19bqh2r"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="primary-filter"
                      type="hidden"
                      value="Bounce Action"
                    />
                  </div>
                </div>
              </div>
              <div
                className="mako-grid__col-9 col-start-4"
              >
                <div
                  className="input-text-wrap"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className="input-text-label"
                  >
                    Description
                  </label>
                  <input
                    aria-describedby=""
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    step={null}
                    type="text"
                    value=""
                  />
                  
                  
                </div>
              </div>
              <i
                className="filter-row-remove sg-icon sg-icon-x"
              />
            </div>
            <div
              className="mako-grid__row"
            >
              <div
                className="row btn-list filter-actions-container"
              >
                <button
                  className="btn btn-small btn-secondary"
                  type="button"
                >
                  Add a Filter
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        data-test="rule-table"
      >
        <table
          className="table-wrap"
        >
          <thead>
            <tr>
              <th
                className=""
                onClick={[Function]}
              >
                Id
              </th>
              <th
                className=""
                onClick={[Function]}
              >
                Bounce Action
              </th>
              <th
                className=""
                onClick={[Function]}
              >
                Response Code
              </th>
              <th
                className=""
                onClick={[Function]}
              >
                Description
              </th>
              <th
                className="actions-cell"
                onClick={[Function]}
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                504
              </td>
              <td>
                no_action
              </td>
              <td>
                550
              </td>
              <td>
                bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl
              </td>
              <td
                className="actions"
              >
                <i
                  className="sg-icon sg-icon-ellipsis"
                  style={null}
                />
                <div
                  className="action-icons"
                >
                  <span
                    data-tooltip="View"
                    data-tooltip-pos="up"
                    onClick={[Function]}
                  >
                    <i
                      className="sg-icon sg-icon-view"
                      style={null}
                    />
                  </span>
                  <span
                    data-tooltip="Edit"
                    data-tooltip-pos="up"
                  >
                    <i
                      className="sg-icon sg-icon-pencil"
                      style={null}
                    />
                  </span>
                  <span
                    data-tooltip="Delete"
                    data-tooltip-pos="up"
                  >
                    <i
                      className="sg-icon sg-icon-trash"
                      style={null}
                    />
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-4 col-start-5"
    >
      <div
        className="pagination pagination-container"
      >
        <a
          className="btn btn-secondary btn-small pagination-prev"
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <Component
            active={false}
            badge={null}
            className="sg-button"
            disabled={false}
            group={false}
            icon={null}
            loading={false}
            onClick={[Function]}
            onDark={false}
            small={false}
            type="secondary"
          >
            Cancel
          </Component>
        </Component>
        <Component
          offset={11}
          width={2}
        >
          <Component
            active={false}
            badge={null}
            className="sg-button"
            disabled={false}
            group={false}
            icon={null}
            loading={false}
            onClick={[Function]}
            onDark={false}
            small={false}
            type="primary"
          >
            Submit
          </Component>
        </Component>
      </Component>
    </div>
  </SideModal>
</div>
`;
